test/Groups/describe.jl

```diff
@@ -0,0 +1,79 @@
+@testset "describe() for permutation groups" begin
+
+  @test describe(symmetric_group(1)) == "1"
+  @test describe(symmetric_group(2)) == "C2"
+  @test describe(symmetric_group(3)) == "S3"
+  @test describe(symmetric_group(4)) == "S4"
+  @test describe(symmetric_group(5)) == "S5"
+
+  @test describe(alternating_group(1)) == "1"
+  @test describe(alternating_group(2)) == "1"
+  @test describe(alternating_group(3)) == "C3"
+  @test describe(alternating_group(4)) == "A4"
+  @test describe(alternating_group(5)) == "A5"
+end
+
+@testset "describe() for pc groups" begin
+  @test describe(dihedral_group(2)) == "C2"
+  @test describe(dihedral_group(4)) == "C2 x C2"
+  @test describe(dihedral_group(6)) == "S3"
+  @test describe(dihedral_group(8)) == "D8"
+  @test describe(dihedral_group(10)) == "D10"
+end
+
+@testset "describe() for matrix groups" begin
+  @test describe(general_linear_group(2,2)) == "S3"         # FIXME: correct but not ideal
+  @test describe(general_linear_group(2,3)) == "GL(2,3)"
+  @test describe(general_linear_group(2,4)) == "GL(2,4)"
+  @test describe(general_linear_group(3,2)) == "PSL(3,2)"   # FIXME: correct but not ideal
+
+  @test describe(special_linear_group(2,2)) == "S3"         # FIXME: correct but not ideal
+  @test describe(special_linear_group(2,3)) == "SL(2,3)"
+  @test describe(special_linear_group(2,4)) == "A5"         # FIXME: correct but not ideal
+  @test describe(special_linear_group(3,2)) == "PSL(3,2)"   # FIXME: correct but not ideal
+  @test describe(special_linear_group(3,3)) == "PSL(3,3)"
+  @test describe(special_linear_group(4,2)) == "A8"         # FIXME: correct but not ideal
+end
+
+@testset "describe() for projective groups" begin
+  # TODO: PGL / projective_general_linear_group constructor is missing
+  # TODO: PSL / projective_general_linear_group constructor is missing
+end
+
+@testset "describe() for free groups" begin
+  @test describe(free_group(0)) == "1"
+  @test describe(free_group(1)) == "Z"
+  @test describe(free_group(2)) == "a free group of rank 2"
+  @test describe(free_group(3)) == "a free group of rank 3"
+
+  F = free_group(4)
+  subs = [sub(F, gens(F)[1:n])[1] for n in 0:4]
+  @test describe(subs[1]) == "1"
+  @test describe(subs[2]) == "Z"
+  @test describe(subs[3]) == "a free group of rank 2"
+  @test describe(subs[4]) == "a free group of rank 3"
+  @test describe(subs[5]) == "a free group of rank 4"
+end
+
+@testset "describe() for finitely presented groups" begin
+
+  F = free_group(1)
+  @test describe(direct_product(F, F)) == "Z x Z"
+
+  F = free_group(2)
+  G = quo(F, [gen(F,2)])[1]
+  @test describe(G) == "Z"
+  G = quo(F, [gen(F,2)/gen(F,1)])[1]
+  @test describe(G) == "Z"
+
+  # TODO: add test for infinitely generated groups
+  @test describe(derived_subgroup(free_group(2))[1]) == "a non-finitely generated group"
+
+end
+
+
+@testset "describe() for some groups in examples" begin
+
+  @test describe(fundamental_group(torus())) == "a finitely presented infinite group"
```

Comment:
To be clear, this is *not* what @micjoswig requested in this case (he'd like something like `Z x Z`).

And I can computed that -- in this case. The problem is to write the code so that it won't get stuck in an infinite loop for other examples. Gotta think a bit about that.

Reply:
Actually I've now added a cheap check that will detect some "obviously" abelian FPGroups, so this example now *is* recognized as `Z x Z`.